src/main/java/seedu/command/EditContactCommand.java

```diff
@@ -43,21 +45,24 @@ public void execute() {
     private Boolean hasDuplicates(Contact postEditContact, ContactList contactList, int contactIndex)
             throws InvalidFlagException {
         ArrayList<Integer> duplicatedIndex = new ArrayList<>();
+        Boolean[] hasEditedField = hasEditedFieldChecker(contactDetails);
```

Comment:
Hi @ashrafjfr , I believe you should be able to use `boolean` as a primitive Java data type. Capitalised `Boolean` is an object (similar to `int` vs `Integer`). Generally, in the case of using `Boolean`, class methods are used. However, in your implementation, I think that `boolean` primitive data type should suffice. In addition, a primitive data type uses a lot less memory than an object.

tl;dr - In your use case, use `boolean` instead of `Boolean`.
```suggestion
        boolean[] hasEditedFields = hasEditedFieldChecker(contactDetails);
```